packages/slice-machine/src/models/libraries/context.tsx

```diff
@@ -9,13 +9,13 @@ import { LibraryUI } from "lib/models/common/LibraryUI";
 import { SliceMockConfig } from "lib/models/common/MockConfig";
 
 export const LibrariesContext = React.createContext<
-  Partial<ReadonlyArray<LibraryState>>
->([]);
+  ReadonlyArray<LibraryState> | undefined
+>(undefined);
```

Comment:
why `| undefined`?

Reply:
That's the goal of this PR, it's in the description,

There are 2 cases:
- the "libraries" key from SM.json is missing and we display a special screen
- the "libraries" key from SM.Json is empty or contains only invalid names, and we display a special screen too.

So here, having it undefined or defined means that we differentiate the two cases.